File: codegen/cOut/java/Checker.java
import com.sun.jna.Pointer;

///A checker on the game board.
class Checker
{
  Pointer ptr;
  int ID;
  int iteration;
  public Checker(Pointer p)
  {
    ptr = p;
    ID = Client.INSTANCE.checkerGetId(ptr);
    iteration = BaseAI.iteration;
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.checkers.length; i++)
    {
      if(BaseAI.checkers[i].ID == ID)
      {
        ptr = BaseAI.checkers[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands

  ///Commands a checker to move to a new location.
  boolean move(int x, int y)
  {
    validify();
    return (Client.INSTANCE.checkerMove(ptr, x, y) == 0) ? false : true;
  }

    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.checkerGetId(ptr);
  }
  ///The players id that controls this Checker.
  public int getOwner()
  {
    validify();
    return Client.INSTANCE.checkerGetOwner(ptr);
  }
  ///The x coordinate of the checker.
  public int getX()
  {
    validify();
    return Client.INSTANCE.checkerGetX(ptr);
  }
  ///The y coordinate of the checker.
  public int getY()
  {
    validify();
    return Client.INSTANCE.checkerGetY(ptr);
  }
  ///1 if kinged,
  public int getKinged()
  {
    validify();
    return Client.INSTANCE.checkerGetKinged(ptr);
  }

}

File: codegen/cOut/csharp/Checker.cs
using System;
using System.Runtime.InteropServices;

/// <summary>
/// A checker on the game board.
/// </summary>
public class Checker
{
  public IntPtr ptr;
  protected int ID;
  protected int iteration;

  public Checker()
  {
  }

  public Checker(IntPtr p)
  {
    ptr = p;
    ID = Client.checkerGetId(ptr);
    iteration = BaseAI.iteration;
  }

  public bool validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.checkers.Length; i++)
    {
      if(BaseAI.checkers[i].ID == ID)
      {
        ptr = BaseAI.checkers[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

  #region Commands
  /// <summary>
  /// Commands a checker to move to a new location.
  /// </summary>
  public bool move(int x, int y)
  {
    validify();
    return (Client.checkerMove(ptr, x, y) == 0) ? false : true;
  }
  #endregion

  #region Getters
  /// <summary>
  /// Unique Identifier
  /// </summary>
  public int Id
  {
    get
    {
      validify();
      int value = Client.checkerGetId(ptr);
      return value;
    }
  }

  /// <summary>
  /// The players id that controls this Checker.
  /// </summary>
  public int Owner
  {
    get
    {
      validify();
      int value = Client.checkerGetOwner(ptr);
      return value;
    }
  }

  /// <summary>
  /// The x coordinate of the checker.
  /// </summary>
  public int X
  {
    get
    {
      validify();
      int value = Client.checkerGetX(ptr);
      return value;
    }
  }

  /// <summary>
  /// The y coordinate of the checker.
  /// </summary>
  public int Y
  {
    get
    {
      validify();
      int value = Client.checkerGetY(ptr);
      return value;
    }
  }

  /// <summary>
  /// 1 if kinged,
  /// </summary>
  public int Kinged
  {
    get
    {
      validify();
      int value = Client.checkerGetKinged(ptr);
      return value;
    }
  }

  #endregion

  #region Properties
  #endregion
}
